src/sail_c_backend/c_backend.ml

```diff
@@ -2452,18 +2452,18 @@ module Codegen (Config : CODEGEN_CONFIG) = struct
       let unit_tests = get_unit_tests cdefs in
 
       let unit_test_functions =
-        ["unit (*const SAIL_TESTS[])(unit) = {"]
+        [sprintf "unit (*const SAIL_TESTS[%d])(unit) = {" (List.length unit_tests + 1)]
         @ List.map (fun id -> sprintf "  %s," (sgen_function_id id)) unit_tests
         @ ["  NULL"; "};"]
         |> separate_map hardline string
       in
 
       let unit_test_names =
-        ["const char* const SAIL_TEST_NAMES[] = {"]
+        [sprintf "const char* const SAIL_TEST_NAMES[%d] = {" (List.length unit_tests + 1)]
         @ Util.map_last
             (fun is_last id -> sprintf "  \"%s\"%s" (String.escaped (string_of_id id)) (if is_last then "" else ","))
```

Comment:
With the trailing `NULL` this `Util.map_last` can just use `List.map` like above, as we don't want to skip the final comma.

Reply:
That's what's breaking the CI:
```
const char* const SAIL_TEST_NAMES[4] = {
  "test_0",
  "test_1",
  "test_2"
  NULL
};
```

Reply:
Why does this need a null terminator, can't we use `sizeof(SAIL_TEST_NAMES) / sizeof(SAIL_TEST_NAMES[0])`;?

Reply:
Hopefully fixed. I also added `SAIL_TEST_COUNT` and used that. I left the `NULL`'s because some people might prefer them in their own test runners, and it makes the trailing comma simpler. Output looks like this now (with no tests):

```

const size_t SAIL_TEST_COUNT = 0;
unit (*const SAIL_TESTS[1])(unit) = {
  NULL
};
const char* const SAIL_TEST_NAMES[1] = {
  NULL
};

void model_test(void)
{
  for (size_t i = 0; i < SAIL_TEST_COUNT; ++i) {
    model_init();
    printf("Testing %s\n", SAIL_TEST_NAMES[i]);
    SAIL_TESTS[i](UNIT);
    printf("Pass\n");
    model_fini();
  }
}
```